### PR TITLE
Add CLI entrypoint

### PR DESCRIPTION
**Pull request by gsheni**
- Add entrypoint to add/override cli commands

### CODE REVIEW

File: featuretools/utils/cli_utils.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import platform
+import struct
+import sys
+
+import pkg_resources
+
+
+# Modified from here
+# https://github.com/pandas-dev/pandas/blob/d9a037ec4ad0aab0f5bf2ad18a30554c38299e57/pandas/util/_print_versions.py#L11
+def get_sys_info():
```
Comment (kmax12) [inline featuretools/utils/cli_utils.py]: can we add a test case that calls `get_installed_packages` and `get_sys_info` even if it doesn’t check the outputs? that would at least catch any runtime errors

File: featuretools/__main__.py [hunk 2]
```diff
@@ -1,8 +1,10 @@
 import click
 import pandas as pd
+import pkg_resources
 
 import featuretools
 from featuretools.primitives.utils import get_featuretools_root
+from featuretools.utils.cli_utils import get_installed_packages, get_sys_info
```
Comment (kmax12) [inline featuretools/__main__.py]: let’s also make them importable from `featuretools.utils` and access them from there. that way if we change the `cli_utils.py` file name it doesnt cause problems

File: featuretools/utils/cli_utils.py (hunk 1 above)
Comment (gsheni) [reply featuretools/utils/cli_utils.py]: Fixed.

File: featuretools/__main__.py (hunk 2 above)
Comment (gsheni) [reply featuretools/__main__.py]: Fixed.

File: featuretools/__main__.py [hunk 3]
```diff
@@ -14,9 +16,36 @@ def cli():
...
+        ("scikit-learn", installed_packages['scikit-learn']),
+        ("pip", installed_packages['pip']),
+        ("setuptools", installed_packages['setuptools']),
+    ]
+    for k, stat in deps:
+        print("{k}: {stat}".format(k=k, stat=stat))
 
-@click.command()
+
+@click.command(name='list-primitives')
```
Comment (gsheni) [inline featuretools/__main__.py]: Replace the underscore with a dash to be more consistent with other CLI tools. 
https://github.com/pallets/click/issues/1123#issuecomment-425167243

**Pull request merged**